### PR TITLE
COMMERCE-10161 Implement Workflow for Product Relationship

### PR DESCRIPTION
**Pull request by alessiorendina**
https://issues.liferay.com/browse/COMMERCE-10161

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_cp_definition_link.jsp [hunk 1]
```diff
@@ -49,5 +62,54 @@ CPDefinition cpDefinition = cpDefinitionLink.getCPDefinition();
...
+			<%
+			String saveButtonLabel = "save";
+
+			if (cpDefinitionLink.isDraft() || cpDefinitionLink.isApproved() || cpDefinitionLink.isExpired() || cpDefinitionLink.isScheduled()) {
+				saveButtonLabel = "save-as-draft";
+			}
+
+			String publishButtonLabel = "publish";
+
+			if (WorkflowDefinitionLinkLocalServiceUtil.hasWorkflowDefinitionLink(themeDisplay.getCompanyId(), scopeGroupId, CPDefinitionLink.class.getName())) {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/edit_cp_definition_link.jsp:91]: can we move this to the context class?

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLinkLocalServiceImpl.java [hunk 2]
```diff
@@ -75,6 +95,29 @@ public CPDefinitionLink addCPDefinitionLinkByCProductId(
 
 		User user = _userLocalService.getUser(serviceContext.getUserId());
 
+		Date expirationDate = null;
+		Date now = new Date();
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLinkLocalServiceImpl.java:99]: I've just seen this https://github.com/brianchandotcom/liferay-portal/pull/134941#issuecomment-1553691954 this should be calld date

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
@alessiorendina, please check if you have other Date now variable

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
@jaydawu remember to link PRs and close the previous one with "ci:close" command. This time I'll do it for you :)

Forwarded here: https://github.com/liferay-commerce/liferay-portal/pull/3813

**Pull request closed without merging**